src/libs/ajax/User.ts

```diff
@@ -0,0 +1,359 @@
+        if (error instanceof Response && (error.status === 404 || error.status === 403)) {
+          return null;
+        }
+        throw error;
+      }
+    },
+
+    getPrivacyPolicy: async (): Promise<string> => {
+      const response = await fetchSam('privacy/text', _.merge(authOpts(), { signal }));
+      return response.text();
+    },
+
+    firstTimestamp: (): Promise<RexFirstTimestampResponse> => {
+      return getFirstTimeStamp(getTerraUser().token!);
+    },
+
+    getNihStatus: async (): Promise<OrchestrationNihStatusResponse | undefined> => {
+      try {
+        const res = await fetchOrchestration('api/nih/status', _.merge(authOpts(), { signal }));
+        return res.json();
+      } catch (error: unknown) {
+        if (error instanceof Response && error.status === 404) {
+          return;
+        }
+        throw error;
+      }
+    },
+
+    linkNihAccount: async (token: string): Promise<OrchestrationNihStatusResponse> => {
+      const res = await fetchOrchestration(
+        'api/nih/callback',
+        _.mergeAll([authOpts(), jsonBody({ jwt: token }), { signal, method: 'POST' }])
+      );
+      return res.json();
+    },
+
+    unlinkNihAccount: async (): Promise<void> => {
+      await fetchOrchestration('api/nih/account', _.mergeAll([authOpts(), { signal, method: 'DELETE' }]));
+    },
+
+    getFenceStatus: async (providerKey: string): Promise<BondFenceStatusResponse | {}> => {
+      try {
+        const res = await fetchBond(`api/link/v1/${providerKey}`, _.merge(authOpts(), { signal }));
+        return res.json();
+      } catch (error: unknown) {
+        if (error instanceof Response && error.status === 404) {
+          return {};
+        }
+        throw error;
+      }
+    },
+
+    getFenceAuthUrl: async (providerKey: string, redirectUri: string): Promise<BondFenceUrlResponse> => {
+      const queryParams = {
+        scopes: ['openid', 'google_credentials', 'data', 'user'],
+        redirect_uri: redirectUri,
+        state: btoa(JSON.stringify({ providerKey })),
+      };
+      const res = await fetchBond(
+        `api/link/v1/${providerKey}/authorization-url?${qs.stringify(queryParams, { indices: false })}`,
+        _.merge(authOpts(), { signal })
+      );
+      return res.json();
+    },
+
+    linkFenceAccount: async (
+      providerKey: string,
+      authCode: string | undefined,
+      redirectUri: string,
+      state: string
+    ): Promise<BondFenceStatusResponse> => {
+      const queryParams = {
+        oauthcode: authCode,
+        redirect_uri: redirectUri,
+        state,
+      };
+      const res = await fetchBond(
+        `api/link/v1/${providerKey}/oauthcode?${qs.stringify(queryParams)}`,
+        _.merge(authOpts(), { signal, method: 'POST' })
+      );
+      return res.json();
+    },
+
+    unlinkFenceAccount: async (providerKey: string): Promise<void> => {
+      return fetchBond(`api/link/v1/${providerKey}`, _.merge(authOpts(), { signal, method: 'DELETE' }));
```

Comment:
Since this is typed as returning a `Promise<void>`, let's not return the Response from void here.
```suggestion
      await fetchBond(`api/link/v1/${providerKey}`, _.merge(authOpts(), { signal, method: 'DELETE' }));
```